### PR TITLE
add vmong5k provider

### LINKED ISSUE #308: support vmong5k provider

### PR DESCRIPTION
**Pull request by jrbalderrama**
fix #308 support vmon5k provider from enoslib

### CODE REVIEW

File: enos/provider/vmong5k.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            }
+
+            machines.append(machine)
+
+    sites = g5k._get_sites(clusters)
+    if len(sites) > 1:
+        raise Exception("Multi-site deployment is not supported yet")
+
+    # name of the network is hard-coded!
+    networks = ["network_interface"]
```
Comment (msimonin) [inline enos/provider/vmong5k.py]: https://github.com/BeyondTheClouds/enos/blob/stable/queens/enos/utils/constants.py#L29

File: enos/provider/vmong5k.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    resources = _configuration.get("resources", {})
+    resources = _configuration.get("topology", resources)
+    for description in extra.gen_enoslib_roles(resources):
+        for group in api.expand_groups(description["group"]):
+            clusters.add(description["flavor"])
+            machine = {
+                "roles": [group, description["role"]],
+                "nodes": description["number"],
+                "cluster": description["flavor"],
+                "min": 1
```
Comment (msimonin) [inline enos/provider/vmong5k.py]: We need to send a `flavour` here.
https://gitlab.inria.fr/discovery/enoslib/blob/master/enoslib/infra/enos_vmong5k/schema.py#L38-41

this depends on the role -- 
I'd say we can choose:
```
control: extra-large
network: large
compute: medium
# anything else: medium
```

available (defauklt flavour): https://gitlab.inria.fr/discovery/enoslib/blob/master/enoslib/infra/enos_vmong5k/constants.py

Comment (jrbalderrama) [reply enos/provider/vmong5k.py]: I also realised that is not strictly required because I could test without defining the flavours.

Comment (msimonin) [reply enos/provider/vmong5k.py]: 1. `nodes` and `min` shouldn't be accepted in the schema (this is a bug in enoslib[1])
2. same as above
3. one should use the advanced description by using a `resources` key under the `provider` key

[1]: https://gitlab.inria.fr/discovery/enoslib/issues/2

File: enos/templates/reservation.yaml.sample [hunk 3]
```diff
@@ -147,7 +164,7 @@ kolla_ref: "stable/queens"
 
 # Vars : globals.yml
 kolla:
+  openstack_release: queens
```
Comment (msimonin) [inline enos/templates/reservation.yaml.sample]: I'll try to fix this in #316 
so I'd say we keep that as it is for now.

File: enos/templates/reservation.yaml.sample [hunk 4]
```diff
@@ -1,12 +1,29 @@
...
+  walltime: '02:00:00'
+
+resources:
+  paravance:
+    network: 1
+    control: 1
+  parasilo:
+    compute: 1
+
+virt_image: /grid5000/virt-images/debian9-x64-base.qcow2
```
Comment (msimonin) [inline enos/templates/reservation.yaml.sample]: hum this should be removed.

File: enos/provider/vmong5k.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    clusters = set()
+    resources = _configuration.get("resources", {})
+    resources = _configuration.get("topology", resources)
+    for description in extra.gen_enoslib_roles(resources):
+        for group in api.expand_groups(description["group"]):
+            clusters.add(description["flavor"])
+            role = description["role"]
+            machine = {
+                "roles": [group, role],
+                "number": description["number"],
```
Comment (msimonin) [inline enos/provider/vmong5k.py]: :+1:

**Pull request merged**